auth/record.go

```diff
@@ -0,0 +1,68 @@
+package auth
+
+import (
+	"bytes"
+	"fmt"
+
+	"github.com/libp2p/go-libp2p/core/record"
+)
+
+// AuthDomain_Nonce is the domain string used for peer records contained in an Envelope.
+const AuthDomain_Nonce = "ww/auth/nonce"
+const nonceSize = 20 // 160 bits
+
+// PeerRecordEnvelopePayloadType is the type hint used to identify peer records in an Envelope.
+// Defined in https://github.com/multiformats/multicodec/blob/master/table.csv
+// with name "libp2p-peer-record".
+var AuthDomain_PayloadType = []byte{0xbb, 0xbb} // TODO:  pick better numbers
+
+var _ record.Record = (*expect)(nil)
```

Comment:
I see the global unamed variable pattern in more than one file, what is it for? 👀